### PR TITLE
PR: Fixes profiler extension and buttons

### LINKED ISSUE #6444: Pressing "Clear comparison" in Profiler while its running stops it and raises a TypeError
## Description
when profiler is running somecode(any code) click "clear comparison". 
the code stop running and spyder pump up an error as following:

>   File "c:\users\53282\appdata\local\programs\python\python36\lib\site-packages\spyder_profiler\widgets\profilergui.py", line 264, in <lambda>
>     self.finished(ec, es))
>   File "c:\users\53282\appdata\local\programs\python\python36\lib\site-packages\spyder_profiler\widgets\profilergui.py", line 326, in finished
>     self.output = self.error_output + self.output
> TypeError: must be str, not NoneType

i found this maybe caused by this:
```python
    def show_data(self, justanalyzed=False):
        if not justanalyzed:
            self.output = None
        self.log_button.setEnabled(self.output is not None \
                                   and len(self.output) > 0)
        self.kill_if_running()
```
so i edit it:
```python
    def show_data(self, justanalyzed=False):
        if not justanalyzed:
            self.output = ''
        self.log_button.setEnabled(self.output is not None \
                                   and len(self.output) > 0)
        #self.kill_if_running()
```
and now "clear comparison" wont stop the code running and no error will pump up.

## Version and main components

* Spyder Version: 3.2.6 
* Python Version: 3.6.4
* Qt Versions: 5.10.0, PyQt5 5.10 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.22.0 (OK)
numpy >=1.7      :  1.14.0 (OK)
sphinx >=0.6.6   :  1.6.7 (OK)
rope >=0.9.4     :  0.10.7 (OK)
jedi >=0.9.0     :  0.11.1 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  None (NOK)
cython >=0.21    :  None (NOK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.2.1 (OK)
pylint >=0.25    :  1.8.2 (OK)

```

### LINKED ISSUE #19633: Bug: FileDialog: Profiler Save/Load does not make File-Extension (.Result) clear
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
The expected extension of the file is unclear if i try to load or save a profiling result.
This could be improved by providing better information, saving the file with extension if no extension was given and/or allowing to select "any file extension" in the save dialog.

This error was found on Gnome and its one of many filedialog integration things, but maybe this can be improved in the future :)

### What steps reproduce the problem?

1. Save Profile/Load Profile
2. See that there is no file extension in the save dialog
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

Python 3.10.4 64-bit | Qt 5.15.2 | PyQt5 5.15.6 | Linux 5.10.0-18-amd64 

image

### Dependencies

```
PASTE DEPENDENCIES HERE
```

### LINKED ISSUE #19965: Cannot upgrade to 5.3.3 from Anaconda Navigator, or by using Conda (Anaconda base environment)
### What steps will reproduce the problem?

If I try to upgrade to 5.3.3 using Anaconda Navigator it fails. If I try to use the terminal and the conda commands, it also fails. Only way I can get 5.3.3 is as stand alone

## Versions

* Spyder version: 5.2.2 
* Python version: 3.9.13
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0;<8.0.0        :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.16.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.4.0 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0                :  2.14.5 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.2;<1.4.0          :  1.3.3 (OK)
pylsp_black >=1.0.0           :  1.0.0 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.2 (OK)
qtpy >=1.5.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  63.4.1 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.2.1;<2.3.0 :  2.2.1 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=17                      :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  0.29.32 (OK)
matplotlib >=2.0.0            :  3.5.2 (OK)
numpy >=1.7                   :  1.21.5 (OK)
pandas >=1.1.1                :  1.4.4 (OK)
scipy >=0.17.0                :  1.9.1 (OK)
sympy >=0.7.3                 :  1.10.1 (OK)
```

### PR DESCRIPTION
**Pull request by maurerle**
## Description of Changes

* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

disables save button if no profile loaded yet
disables load and clear button while profiling is running
add default extension if no extension given on save dialog

### Issue(s) Resolved

Fixes #19633 
Fixes #6444

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Florian Maurer

### CODE REVIEW

File: spyder/plugins/profiler/widgets/main_widget.py [hunk 1]
```diff
@@ -700,6 +711,7 @@ def initialize_view(self):
         self.item_list = []  # To be use for collapsing/expanding one level
         self.items_to_be_shown = {}
         self.current_view_depth = 0
+        #self.stats1 = []
```
Comment (dalthviz) [inline spyder/plugins/profiler/widgets/main_widget.py:714]: Maybe this line should be removed?

File: spyder/plugins/profiler/widgets/main_widget.py [hunk 2]
```diff
@@ -738,7 +750,8 @@ def hide_diff_cols(self, hide):
 
     def save_data(self, filename):
         """Save profiler data."""
-        self.stats1[0].dump_stats(filename)
+        if len(self.stats1) >0:
```
Comment (dalthviz) [inline spyder/plugins/profiler/widgets/main_widget.py:753]: Missing space after operator
```suggestion
        if len(self.stats1) > 0:
```

**Review by dalthviz — COMMENTED** [review]
Thanks @maurerle for the help with this! I did an edit to the original PR comment (it was pointing to a different issue). Also, I left a couple of comments/suggestions regarding a commented out line and a missing space after an operator.

Could it be possible for you to upload a GIF or video showing the new behavior?

I'm missing doing a check locally but other than that this LGTM :+1:

**Comment by maurerle** [comment]
Thank you  @dalthviz,

I attached a video showing how the buttons are disabled.
Saving the last report while running a report is still possible without issues (saves previous results)
But loading and clearing would make issues, so they are disabled while the profiler is running.

After saving the file, one can see that the correct .Result extension is added (when loading results for comparison at the end)

[spyder-profiler-screencast.webm](https://user-images.githubusercontent.com/25026204/202273476-cfc1316d-7aac-4415-b5f2-e36378a8893f.webm)

Thank you, I also fixed the fixes issues  to the two correct ones :)

**Review by dalthviz — APPROVED** [review]
Thanks @maurerle for addressing the comments so fast, the video preview of how things work with the changes here and setting the correct issue number on the comment :)  ! Seems like this is working also on Windows so this LGTM :+1:

Note:

The failing checks related to the conda-based installers are unrelated to the changes on this PR

**Pull request merged**